engine/access/rest/websockets/data_providers/blocks_provider.go

```diff
@@ -88,8 +88,11 @@ func (p *BlocksDataProvider) Run() error {
 				return nil, fmt.Errorf("failed to build block response :%w", err)
 			}
 
-			var response models.BaseDataProvidersResponse
-			response.Build(p.ID(), p.Topic(), &block)
+			response := models.BaseDataProvidersResponse{
+				SubscriptionID: p.ID(),
+				Topic:          p.Topic(),
+				Payload:        &block,
+			}
 
 			return &response, nil
```

Comment:
consider this for here and elsewhere
```suggestion
			return &models.BaseDataProvidersResponse{
				SubscriptionID: p.ID(),
				Topic:          p.Topic(),
				Payload:        &block,
			}, nil
```

Reply:
I intentionally kept it this way to improve readability. Naming the variable (`response` in this case) makes the code clearer, as it explicitly conveys what the `interface{}` value represents. Without it, a developer would need to infer the meaning from the struct itself. This small addition reduces cognitive load and makes the code more self-explanatory.
What do you think of it?